omod/src/main/java/org/openmrs/module/msfcore/formaction/LabOrderFormAction.java

```diff
@@ -19,15 +21,22 @@
     private MSFCoreService msfCoreService;
 
     /**
-     * Generate test orders when request investication section of baseline form is submited
+     * Generate test orders when request investication section of baseline form
+     * is submited
      */
     @Override
 	public void apply(String operation, FormEntrySession session) {
-		if (session.getForm().getUuid().equals(MSFCoreConfig.HTMLFORM_REQUEST_INVESTIGATION_UUID)) {
+
+		boolean isRequestInvestigationForm = Arrays
+				.asList(MSFCoreConfig.HTMLFORM_REQUEST_INVESTIGATION_UUID,
+						MSFCoreConfig.FORM_NCD_FOLLOWUP_REQUEST_INVESTIGATION_UUID)
+				.contains(session.getForm().getUuid());
+
+		if (isRequestInvestigationForm) {
+
 			Visit visit = session.getEncounter().getVisit();
-			encounterService.getEncountersByVisit(visit, false).stream()
-			        .filter(e -> e.getForm().getUuid().equals(MSFCoreConfig.HTMLFORM_REQUEST_INVESTIGATION_UUID)).findFirst()
-			        .ifPresent(e -> msfCoreService.saveTestOrders(e));
+			encounterService.getEncountersByVisit(visit, false).stream().filter(e -> isRequestInvestigationForm)
```

Comment:
Guimino, we need to generate the test orders only for the encounters of **request investigation** so the filter must return that one only. Something like the following.

```suggestion
			encounterService.getEncountersByVisit(visit, false).stream().filter(e -> e.getForm().getUuid().equals(session.getForm().getUuid()))
```

Reply:
Ok @edrisse.I have changed  as your suggestions. Thanks